lib/modules/manager/hermit/extract.ts

```diff
@@ -0,0 +1,103 @@
+import is from '@sindresorhus/is';
+import minimatch from 'minimatch';
+import upath from 'upath';
+import { logger } from '../../../logger';
+import { readLocalDirectory } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import { HermitDatasource } from '../../datasource/hermit';
+import type { PackageDependency, PackageFile } from '../types';
+import type { HermitListItem } from './types';
+
+const pkgReferenceRegex = regEx(`(?<packageName>.*?)-(?<version>[0-9]{1}.*)`);
+
+/**
+ * extractPackageFile scans the folder of the package files
+ * and looking for .{packageName}-{version}.pkg
+ */
+export async function extractPackageFile(
+  content: string,
+  filename: string
+): Promise<PackageFile | null> {
+  logger.trace('hermit.extractPackageFile()');
+  const dependencies = [] as PackageDependency[];
+  const packages = await listHermitPackages(filename);
+
+  if (packages === null) {
```

Comment:
```suggestion
  if (!packages?.length) {
```

return null if empty